packages/ilios-common/addon/components/new-session.js

```diff
@@ -1,27 +1,21 @@
 import Component from '@glimmer/component';
 import { service } from '@ember/service';
-import { cached, tracked } from '@glimmer/tracking';
+import { tracked } from '@glimmer/tracking';
 import { action } from '@ember/object';
 import { dropTask } from 'ember-concurrency';
-import { validatable, Length, NotBlank } from 'ilios-common/decorators/validation';
 import { findBy } from 'ilios-common/utils/array-helpers';
-import { TrackedAsyncData } from 'ember-async-data';
+import YupValidations from 'ilios-common/classes/yup-validations';
+import { string } from 'yup';
 
-@validatable
 export default class NewSessionComponent extends Component {
   @service store;
 
-  @NotBlank() @Length(3, 200) @tracked title;
+  @tracked title;
   @tracked selectedSessionTypeId;
 
-  @cached
```

Comment:
I always wondered how the `@cached` decorator worked for async data. Now that it's not being used, does that mean that the validation being done is no longer cached? Is that bad?

Reply:
The data point under validation, `title`, is at no point in an async state. but since the `hasErrorFor` function is async, we currently have to do this dance in wrapping it. 
Getting rid of all this extra code is a huge improvement for the DX, at least. IMO.

Reply:
The validation is still an asynchronous process (because you can do asynchronous stuff in there, we're just not doing it here), however what happens now is that we-revalidate constantly if there are errors they get stored and then we display them, they can also go away, just constant shuffle, but all the async is elsewhere. The only part to worry about is validating before actually doing anything and that's where `isValid()` comes in. When you run that and wait for results you get a settled correct state.